pkg/resourcemanager/webhook/highavailabilityconfig/handler.go

```diff
@@ -460,6 +470,45 @@ func (h *Handler) mutateTopologySpreadConstraints(
 	}
 }
 
+func (h *Handler) mutatePodTolerationSeconds(podTemplateSpec *corev1.PodTemplateSpec) {
+	var (
+		toleratesNodeNotReady    bool
+		toleratesNodeUnreachable bool
+	)
+
+	// Check if toleration is already specific in podTemplate.
+	for _, toleration := range podTemplateSpec.Spec.Tolerations {
+		if len(toleration.Effect) > 0 && toleration.Effect != corev1.TaintEffectNoExecute {
+			continue
+		}
+
+		switch toleration.Key {
+		case corev1.TaintNodeNotReady:
+			toleratesNodeNotReady = true
+		case corev1.TaintNodeUnreachable:
+			toleratesNodeUnreachable = true
+		}
```

Comment:
What if they are already tolerated (which I believe should be the standard situation, since the mutating webhook admission plugin in KAPI runs at the very end (after the in-tree plugins), hence https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/defaulttolerationsecond has already "defaulted" the values)? Shouldn't we overwrite them then to our desired value?

Reply:
The admission plugin acts on `Pod`s while our HA-Config webhook mutates `Deployment`s and `StatefulSet`s, so we are fine in this case.

Reply:
:) Good point, thanks for clarifying!